File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/models/BasicGroupModel.java
package com.inspiredandroid.linuxcommandbibliotheca.models;

import com.inspiredandroid.linuxcommandbibliotheca.R;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

/**
 * Created by simon on 19.04.17.
 */

public class BasicGroupModel extends RealmObject {

    @PrimaryKey
    private int id;
    private int position;
    private String title;
    private RealmList<CommandGroupModel> groups;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public RealmList<CommandGroupModel> getGroups() {
        return groups;
    }

    public void setGroups(RealmList<CommandGroupModel> groups) {
        this.groups = groups;
    }

    public int getIconResourceId() {
        switch (id) {
            case 247:
                return R.drawable.ic_user_male_circle_40dp;
            case 37:
                return R.drawable.ic_folder_black_40dp;
            case 246:
                return R.drawable.ic_info_40dp;
            case 21:
                return R.drawable.ic_settings_black_40dp;
            case 43:
                return R.drawable.ic_video_trimming_40dp;
            case 243:
                return R.drawable.ic_network_card_40dp;
            case 179:
                return R.drawable.ic_search_40dp;
            case 248:
                return R.drawable.ic_skull_40dp;
            case 249:
                return R.drawable.ic_package_40;
            case 29:
                return R.drawable.ic_videogame_asset_black_40dp;
            case 250:
                return R.drawable.ic_1_circle_40dp;
        }
        return R.drawable.ic_user_male_circle_40dp;
    }
}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/models/BasicGroupModel.kt
package com.inspiredandroid.linuxcommandbibliotheca.models

import com.inspiredandroid.linuxcommandbibliotheca.R

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required

/**
 * Created by simon on 19.04.17.
 */

open class BasicGroupModel : RealmObject() {

    @PrimaryKey
    var id: Int = 0
    var position: Int = 0
    @Required
    var title: String? = null
    var groups: RealmList<CommandGroupModel>? = null

    val iconResourceId: Int
        get() {
            when (id) {
                247 -> return R.drawable.ic_user_male_circle_40dp
                37 -> return R.drawable.ic_folder_black_40dp
                246 -> return R.drawable.ic_info_40dp
                21 -> return R.drawable.ic_settings_black_40dp
                43 -> return R.drawable.ic_video_trimming_40dp
                243 -> return R.drawable.ic_network_card_40dp
                179 -> return R.drawable.ic_search_40dp
                248 -> return R.drawable.ic_skull_40dp
                249 -> return R.drawable.ic_package_40
                29 -> return R.drawable.ic_videogame_asset_black_40dp
                250 -> return R.drawable.ic_icons8_hand_with_pen
                251 -> return R.drawable.ic_vpn_key_black_24dp
                252 -> return R.drawable.ic_icons8_github
            }
            return R.drawable.ic_user_male_circle_40dp
        }
}
